var/spack/repos/builtin/packages/fplo/package.py

```diff
@@ -0,0 +1,144 @@
+    manual_download = True
+
+    maintainers = ["glennpj"]
+
+    version("22.00-62", sha256="0d1d4e9c1e8e41900901e26c3cd08ee39dcfdeb3f2c4c8862055eaf704b6d69e")
+
+    patch("lapackconfig.patch")
+    patch("MMakefile.patch")
+    patch("ncurses.patch")
+    patch("qt-make.patch")
+    patch("fedit_py.patch")
+
+    depends_on("mkl")
+    depends_on("ncurses")
+    depends_on("perl", type="run")
+    depends_on("qt@5+opengl")
+
+    extends("python")
+    depends_on("py-numpy")
+
+    conflicts("%gcc@12:")
+
+    @property
+    def build_directory(self):
+        return join_path(self.stage.source_path, "FPLO{0}".format(self.version))
+
+    def edit(self, spec, prefix):
+        # Need to set this to 'gcc' even if using the intel compiler as all of
+        # the configuration files are named with 'gcc'.
+        if platform.system() == "Linux":
+            fplo_cc = "gcc"
+        else:
+            fplo_cc = os.path.basename(self.compiler.cc)
+
+        fplo_fc = os.path.basename(self.compiler.fc)
+
+        conffile = "{0}-{1}-{2}-{3}".format(
+            fplo_cc, fplo_fc, platform.system(), platform.machine()
+        )
+        mmakefile = FileFilter(join_path(self.build_directory, "install", "MMakefile"))
+        mmakefile.filter(r"(^conffile=\$configdir/)$", r"\1{0}".format(conffile))
+        mmakefile.filter(r"(^mkl=).*", r"\11")
+
+        # use spack compiler
+        files = glob.glob(join_path(self.build_directory, "install", "conf", "*"))
+        filter_file(r"^\s*CC\s*=.*", "CC=" + spack_cc, *files)
+        filter_file(r"^\s*CXX\s*=.*", "CXX=" + spack_cxx, *files)
+        filter_file(r"^\s*F90\s*=.*", "F90=" + spack_fc, *files)
+
+        # patch for 64 bit integers
+        if "^mkl+ilp64" in spec:
+            setuphelper = FileFilter(join_path(self.build_directory, "PYTHON", "setuphelper.py"))
+            setuphelper.filter("mkl 64bit integer 32bit", "mkl 64bit integer 64bit")
+
+        # setup python build
+        python_makefile = FileFilter(join_path(self.build_directory, "PYTHON", "Makefile"))
+        python_makefile.filter(r"(build_ext\s* --inplace)\s*--interactive(\s*.*)", r"\1\2")
+
+    def build(self, spec, prefix):
+        mmakefile = Executable(join_path(self.build_directory, "install", "MMakefile"))
+        mmakefile_args = [
+            "-f90",
+            spack_fc,
+            "-cc",
+            spack_cc,
+            "-c+",
+            spack_cxx,
+        ]
+
+        with working_dir(self.build_directory):
+            # copy contents of bin
+            copy_tree("bin", join_path(self.stage.source_path, "bin"))
+
+            # configure the build
+            with working_dir("install"):
+                mmakefile(*mmakefile_args)
+
+            # build main
+            make()
+            make("install")
+
+            # build XFBP
+            with working_dir(join_path("XFBP_rel", "XFBP")):
+                make()
+                make("install")
```

Comment:
Would it work to divide `build` and `install` and only pass `install` targets with `-C`?

Reply:
@alalazo Do you mean, as opposed to setting `with working_dir()` multiple times?

Reply:
The native build system for this is not setup to recurse into the various directories, and there is no top level `install` target. It essentially installs the components into the top level (source) directory. I then use Spack to move the desired components into `prefix` to separate it from the source tree.

Reply:
Ok, thanks for clarifying that.